examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+    model, X_train, y_train, cv=cv, return_estimator=True, n_jobs=2
+)
+
+# %%
+# Here, we used a repeated K-fold cross-validation. At each round of
+# cross-validation, it should be noted that the parameter `alpha` of the ridge
+# regressor is also optimized via another internal cross-validation. This
+# process is called a nested cross-validation and should always be implemented
+# whenever model's parameters need to be optimized.
+#
+# Analyze model performance
+# .........................
+#
+# As previously mentioned, one should look at the variance of the model
+# performance within the cross-validation framework.
+import matplotlib.pyplot as plt
+
+cv_score = cv_results["test_score"]
+plt.hist(cv_score, bins=200, density=True)
+plt.xlim([0, 1])
+plt.ylabel("Density")
+plt.xlabel("R2 score")
+_ = plt.title(
+    "Distribution of the scores on the test sets\n during the cross-validation"
+)
+
+# %%
+# We start by plotting the empirical distribution of the test score computed
+# during cross-validation.
+#
+# We observe little variation in R2 score and can thus safely use the results
+# to interpret the model performance.
+#
+# Our baseline model has an R2 of around 0.6 which is better than a dummy
+# regressor. We can therefore use this predictive model as a baseline against
+# more advanced machine learning pipelines.
+#
+# To conclude, cross-validation allows us to answer to two questions: are the
+# results reliable and, if it is, how good is the predictive model.
+#
+# Model inspection
+# ................
+#
+# Once we are happy or at least aware of any model limitations, we can
+# investigate the internals of our model. When we performed the
+# cross-validation, we set the parameter `return_estimator` to `True`.
+#
+# It allows us to get the different predictive models trained and tested within
+# cross-validation.
+cv_estimators = cv_results["estimator"]
+
+# %%
+# While the cross-validation allows us to know if our models are reliable, we
+# have not checked if the model at each cross validation fold is similar.
+#
+# Recall that in our case, our baseline model required tuning of the parameter
+# `alpha`. Therefore, up-to-now, we do not know if the optimal `alpha`
+# parameter for all models are similar. In other words, we are asking the
+# question: what is the variance of the `alpha` parameter value across
+# iterations.
+#
+# Indeed, if the `alpha` parameters vary depending on the input data, it
+# might be challenging to put our model in production because we would not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter, `alpha`, values are centered and
+# condensed around 40. This is a good sign and means that most of the models
```

Comment:
I find the plot not that much convincing as there are several values of alpha far beyond 40.